cpp/modmesh/buffer/BufferExpander.hpp

```diff
@@ -91,11 +91,15 @@ class BufferExpander
     BufferExpander & operator=(BufferExpander &&) = delete;
     ~BufferExpander() = default;
 
+    explicit operator bool() const noexcept { return bool(m_data_holder); }
+
     size_type size() const noexcept
     {
         return static_cast<size_type>(this->m_end - this->m_begin);
     }
 
+    size_t nbytes() const noexcept { return size(); }
```

Comment:
maybe we should remove `nbytes` in both `ConcreteBuffer` and `BufferExpander`, it's redundant to `size()`

Reply:
`nbytes()` is intuitive.  It's good redundancy.  It's hard to tell `size()` returns the number of bytes in the buffer managers.

The redundancy also works as document.  A maintainer immediately knows that `size()` is number of bytes by reading the alias.